stan/math/opencl/matrix_cl.hpp

```diff
@@ -377,8 +435,9 @@ class matrix_cl<T, require_arithmetic_t<T>> {
     try {
       buffer_cl_ = cl::Buffer(ctx, CL_MEM_READ_WRITE, sizeof(T) * A.size());
       cl::Event transfer_event;
-      queue.enqueueWriteBuffer(buffer_cl_, CL_FALSE, 0, sizeof(T) * A.size(),
-                               A.data(), nullptr, &transfer_event);
+      queue.enqueueWriteBuffer(
+          buffer_cl_, std::is_rvalue_reference<Vec&&>::value, 0,
```

Comment:
```suggestion
          buffer_cl_,  opencl_context.in_order() || std::is_rvalue_reference<Vec&&>::value, 0,
```

Reply:
This was not completely fixed. You skipped `opencl_context.in_order() || ` part.